[mlir][vector] Remove unneeded mask restriction

jpienaar opened the pull request:
These were added when the only mapping was to LLVM.

dcaballe commented on mlir/include/mlir/Dialect/Vector/IR/VectorOps.td:
```diff
@@ -1819,17 +1819,17 @@ def Vector_MaskedLoadOp :
-    Results<(outs VectorOfRank<[1]>:$result)> {
+               VectorOf<[I1]>:$mask,
+               AnyVector:$pass_thru)>,
+    Results<(outs AnyVector:$result)> {
```
Probably good to add constraints to ensure that pass_thru and result's types match and that the number of elements in mask and one of the other vectors match as well.

jpienaar replied:
Ha this let to finding a bug in the generated inference with variadics (well known limitations there)

dcaballe commented on mlir/include/mlir/Dialect/Vector/IR/VectorOps.td:
```diff
@@ -2140,14 +2140,14 @@ def Vector_CompressStoreOp :
-               VectorOfRankAndType<[1], [I1]>:$mask,
-               VectorOfRank<[1]>:$valueToStore)> {
+               VectorOf<[I1]>:$mask,
+               AnyVector:$valueToStore)> {
```
For compress and expand, I would state in the doc that the compression/expansion applies only along the innermost dimension.

jpienaar replied:
SG

dcaballe reviewed (commented):
Thanks, Jacques!

jpienaar committed:
[mlir][vector] Remove unneeded restrictions

These were created when the only mapping was to LLVM, rather than some intrinsic property of the op.
